File: frontend/mobile/lib/models/requests/hold_sector_request.dart
class HoldSectorRequest {
  final int quantity;
  /// DailyEntry only — `yyyy-MM-dd`, must fall within the Sector's PeriodYear/PeriodMonth.
  final String? date;

  const HoldSectorRequest({required this.quantity, this.date});

  Map<String, dynamic> toJson() => {
        'quantity': quantity,
        if (date != null) 'date': date,
      };
}

File: backend/services/ticketing/eTicketing.Ticketing.Business/Sectors/Dtos/HoldSectorRequest.cs
namespace eTicketing.Ticketing.Business.Sectors;

public record HoldSectorRequest
{
    public int Quantity { get; init; } = 1;

    // Required iff the sector's TicketingMode is DailyEntry — which exact calendar date within
    // the sector's PeriodYear/PeriodMonth the buyer wants tickets for.
    public DateOnly? Date { get; init; }
}
